crates/cxx-qt-gen/src/parser/qobject.rs

```diff
@@ -73,19 +73,33 @@ impl ParsedQObject {
         let has_qobject_macro = attribute_take_path(&mut declaration.attrs, &["qobject"]).is_some();
 
         // Find if there is any base class
+        // let base_class = attribute_take_path(&mut declaration.attrs, &["base"])
+        //     .map(|attr| {
+        //         let string = expr_to_string(&attr.meta.require_name_value()?.value)?;
+        //         if string.is_empty() {
+        //             // Ensure that the base class attribute is not empty, as this is not valid in both cases
+        //             // - when there is a qobject macro it is not valid
+        //             // - when there is not a qobject macro it is not valid
+        //             return Err(Error::new_spanned(
+        //                 attr,
+        //                 "The #[base] attribute cannot be empty!",
+        //             ));
+        //         }
+        //         Ok(string)
+        //     })
+        //     .transpose()?;
+
         let base_class = attribute_take_path(&mut declaration.attrs, &["base"])
-            .map(|attr| {
-                let string = expr_to_string(&attr.meta.require_name_value()?.value)?;
-                if string.is_empty() {
-                    // Ensure that the base class attribute is not empty, as this is not valid in both cases
-                    // - when there is a qobject macro it is not valid
-                    // - when there is not a qobject macro it is not valid
-                    return Err(Error::new_spanned(
-                        attr,
-                        "The #[base] attribute cannot be empty!",
-                    ));
+            .map(|attr| -> Result<Ident> {
+                let expr = &attr.meta.require_name_value()?.value;
+                if let Expr::Path(path_expr) = expr {
+                    Ok(path_expr.path.require_ident()?.clone())
+                } else {
+                    Err(Error::new_spanned(
+                        expr,
+                        "Base must be a identifier and cannot be empty!",
+                    ))
```

Comment:
somewhere, unsure where? Might even have to be later in the generator phase where it does that fake trait. We need to ensure that whatever type is used in the `#[base_class = T]` has been defined as a type and known to the TypeNames.

eg
```rust
extern "RustQt" {
   #[qobject]
   #[base_class = T]
   type MyObject = ...;
}
```

should fail without having a block that also defines the T
```rust
extern "C++" {
   include!("t.h");
   type T;
}
```